src/averageAges.js

```diff
@@ -55,7 +75,23 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  let filteredByChildren = people.filter(
+    x => people.some(c => c.name === x.mother));
+
+  filteredByChildren = onlyWithSon
+  ? filteredByChildren.filter(
+    x => x.sex === 'm')
+  : filteredByChildren;
+
+  const ageDifferencesArray
+    = filteredByChildren.map(
+      x => x.born - people.find(
+        c => c.name === x.mother
+      ).born
+    );
```

Comment:
1) no need to add `...Array` postfix at the end, `...s` post refers to array itself
2) also, here we need our `getMother` function
```suggestion
  const ageDifferences = children.map(child => {
    const mother = getMother(mother);
    
    return child.born - mother.born;   
  });
```